internal/devbox/providers/nixcache/nixcache.go

```diff
@@ -106,7 +216,56 @@ func checkIfUserCanAddSubstituter(ctx context.Context) {
 			os.Stderr,
 			"In order to use a custom nix cache you must be a trusted user. Please "+
 				"add your username to nix.conf (usually located at /etc/nix/nix.conf)"+
-				" and restart the nix daemon.",
+				" and restart the nix daemon.\n",
 		)
 	}
 }
+
+// Credentials are short-lived credentials that grant access to a private Nix
+// cache.
+type Credentials struct {
+	aws awsCredentials
+}
+
+// Env returns the credentials as a slice of environment variables.
+func (c Credentials) Env() []string {
+	return []string{
+		"AWS_ACCESS_KEY_ID=" + c.aws.AccessKeyID,
+		"AWS_SECRET_ACCESS_KEY=" + c.aws.SecretAccessKey,
+		"AWS_SESSION_TOKEN=" + c.aws.SessionToken,
+	}
+}
+
+func (c Credentials) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.aws)
+}
+
+func (c *Credentials) UnmarshalJSON(data []byte) error {
+	if c == nil {
+		c = &Credentials{}
+	}
+	return json.Unmarshal(data, &c.aws)
+}
```

Comment:
If we embed the aws credentials in the struct we won't need these.

Reply:
The intent was to hide the raw AWS credentials and the Version field, but I can consolidate.